custom/icds_reports/tasks.py

```diff
@@ -1752,8 +1770,12 @@ def get_data_not_in_ucr(status_record):
     chunk_size = 1000
     for chunk in chunked(matching_records_for_db, chunk_size):
         doc_ids = [val[0] for val in chunk]
-        doc_id_and_inserted_in_ucr = _get_docs_in_ucr(domain, status_record.table_id, doc_ids)
+        known_bad_doc_ids = set(
+            InvalidUCRData.objects.filter(doc_id__in=doc_ids).values_list('doc_id', flat=True))
```

Comment:
I think this might be way slower. This replaces one query at the end, only for the missing doc ids, to one query per thousand docs, each with 1000 ids. That goes from one db query to thousands per run.

Reply:
Very good point, reverting this change.